plumpy/processes.py

```diff
@@ -56,19 +51,7 @@ class BundleKeys(object):
     OUTPUTS = 'OUTPUTS'
 
 
-# Use thread-local storage for the stack
-_thread_local = threading.local()  # pylint: disable=invalid-name
-
-
-def _process_stack():
-    """Access the private live stack"""
-    global _thread_local  # pylint: disable=global-statement, invalid-name
-    # Lazily create the first time it's used
-    try:
-        return _thread_local.process_stack
-    except AttributeError:
-        _thread_local.process_stack = []
-        return _thread_local.process_stack
+_process_stack = ContextVar('process stack', default=[])  # pylint: disable=invalid-name
```

Comment:
I need to have a further look at this, I have some concerns from last time I looked at it whether it can do what we want.